### ISSUE TITLE
`{{jsxref("Number")}}` isn't a number primitive but a wrapper object

### ISSUE THREAD

**Issue by mfuji09** [mdn/content#22581]
### MDN URL

https://developer.mozilla.org/en-US/docs/Glossary/Falsy

### What information was incorrect, unhelpful, or incomplete?

`{{jsxref("Number")}}` produces a link to the wrapper object rather than a number primitive, but it is used wrongly in the table.
Actually, `new Number(0) ? 'true' : 'false'` results 'true' though it is written in the table that {{jsxref("Number")}} zero is falsy.

I think the all instances of `{{jsxref("Number")}}` which shows a number primitive should be replaced with 'number'.

### What did you expect to see?

The number zero (so, also `0.0`, etc., and `0x0`). etc.

### Do you have anything more you want to share?

There are similar problems for `Boolean` and `String`.

**Comment by Josh-Cena** [comment]
@mfuji09 As of today, after some reforms, the global object pages also talk about the primitive's semantics. For example, `Number` also includes information about IEEE-754 encoding, number primitive coercion, etc. As such, I would worry less about linking to them where we are referring to the primitive type. If you wish, you can remove the code typography, like:

> The [number](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number) zero (so, also 0.0, etc., and 0x0).

Which is what I do in some other places.

**Comment by mfuji09** [comment]
@Josh-Cena 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number
This page ALSO talks number primitives, but I think it is considered as a page for the `Number` wrapper object, because it remains under the "Global_Objects" directory.

If you worry to be decreased links, shouldn't it be replaced with another links, for example:
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Data_structures#number_type
- https://developer.mozilla.org/en-US/docs/Glossary/Number

**Comment by Josh-Cena** [comment]
So there are the following reasons why I prefer linking to `Global_Objects`.

1. We can't link to `Glossary`, because we don't have `Glossary/Symbol` anymore, so we have to link to the JS docs anyway. Plus, glossary entries are really short and do little more than redirecting readers to the reference.
2. `Data_structures` is primarily a guide article and doesn't contain nearly as much information as the respective reference. Even in the article itself, each section contains a link to the corresponding reference of the global object.
3. `Global_Objects` doesn't necessarily mean "the wrapper object". For `BigInt` and `Symbol`, the distinction between wrapper objects and primitive is nearly irrelevant—it's so hard to get a wrapper object that if you link to `Global_Objects/BigInt` readers are unambiguously going to understand that it refers to the primitive value. It's fair if we propagate this same logic to other "wrapper object" pages, and talk more about the primitive values' semantics there.

TL;DR: if there are three candidate pages to link to: A, B, and C, where A has a sentence saying "for more information see B and C", B has a sentence saying "for more information see C", while C is really the source of all truths, then we should definitely link to C directly.

**Comment by mfuji09** [comment]
@Josh-Cena 
Although `Global_Objects` directory is not only for wrapper objects, it is obviously for objects. So I don't think `Global_Objects` is a sutiable place for explaining primitives (which is not a object).

I think `Data_structures` page is better and incorrect links to `Global_Objects` of the page should be fixed.

**Comment by Josh-Cena** [comment]
We have fully onboarded the decision that pages such as `Number` and `String` describe primarily the primitive types because there's really no point in describing wrapper objects (other than they are bad), and `Number()` and `String()` as functions create primitives, while static methods such as `Number.isNaN`, etc. all work with primitives. I have given more detailed reasons why this is expected and few readers should interpret `Global_Objects` as "just objects". I'm going to close this, and in the future `jsxref("Number")` and `Glossary("Number")` can both safely be used to represent the number type, although the former is strongly recommended.
